packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -606,129 +539,37 @@ class RawMenuAnchor extends StatelessWidget {
     ]
   );
 
-  /// The overlay builder used by the default [RawMenuAnchor] constructor.
-  ///
-  /// The [defaultOverlayBuilder] constructor builds a simple menu overlay. An
-  /// internal [FocusScope] is created to manage focus. Default keyboard
-  /// shortcuts include:
-  ///
-  /// - `ArrowUp` moves focus to the previous menu item.
-  /// - `ArrowDown` moves focus to the next menu item.
-  /// - `ArrowLeft` moves focus out of a submenu, thereby closing that submenu.
-  /// - `ArrowRight` opens and moves focus into a submenu.
-  /// - `Home` moves focus to the first menu item.
-  /// - `End` moves focus to the last menu item.
-  /// - `Escape` closes all menu layers.
-  ///
-  /// To customize the overlay, use the [RawMenuAnchor.overlayBuilder]
-  /// constructor.
-  Widget defaultOverlayBuilder(
-    BuildContext context,
-    List<Widget> menuChildren,
-    RawMenuAnchorOverlayPosition position,
-  ) {
-    return _MenuOverlay(
```

Comment:
I also shuffled the `defaultOverlayBuilder` around a bit (`_MenuOverlay` and `_MenuOverlayPanel` are no longer separate widgets).

**Before**
```dart
class RawMenuAnchor {
  Widget defaultOverlayBuilder() {
    return _MenuOverlay(…);
  }
}

class _MenuOverlay {
  Widget build() {
    return ConstrainedBox(
      // ...
      child: _MenuOverlayPanel(…),
    );
  }
}

class _MenuOverlayPanel {
  Widget build() {
    return UnconstrainedBox(…);
  }
}
```

**After**
```dart
class RawMenuAnchor {
  const factory RawMenuAnchor(…) = _RawMenuAnchorDefaultOverlay;
}

class _RawMenuAnchorDefaultOverlay {
  Widget _buildOverlay() {
    return ConstrainedBox(
      // ...
      child: UnconstrainedBox(…),
    );
  }
}
```